src/puppetlabs/puppetdb/schema.clj

```diff
@@ -50,12 +50,12 @@
 (defn blacklist->vector
   "Take a facts list as either a comma seperated string
    or a sequence and return a vector of those facts"
-  [fact-list]
+  [fl]
   (cond
-    string? (->> (str/split fact-list #",")
-                 (map str/trim)
-                 (apply vector))
-    sequential? (vec fact-list)
+    (string? fl) (->> (str/split fl #",")
+                      (map str/trim)
+                      (apply vector))
+    (and (coll? fl) (every? string? fl)) (vec fl)
```

Comment:
Not that it really matters, but do we really need it to be a vector instead of say a list or whatever?

Reply:
We expect a vector in the schema conversion/validation stuff so this has to return a vector. We could change the validation code on a second pass at this but for now it should be good.

Reply:
No, that reason's plenty good.